### PR TITLE
ENG-12062: AAUIW Adhoc planning problems no longer print java excepti…

### PR DESCRIPTION
**Pull request by mreading**
…on names.

### CODE REVIEW

File: src/frontend/org/voltdb/compiler/PlannerTool.java [hunk 1]
```diff
@@ -158,7 +158,10 @@ public synchronized CompiledPlan planSqlCore(String sql, StatementPartitioning p
                 logException(e, "Error compiling query");
                 loggedMsg = " (Stack trace has been written to the log.)";
             }
-            throw new RuntimeException("Error compiling query: " + e.toString() + loggedMsg, e);
+            if (e instanceof NumberFormatException) {
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/compiler/PlannerTool.java]: Why is `NumberFormatException` special?

File: tests/sqlcmd/baselines/simple/parsedates.errbaseline [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
-Unexpected Ad Hoc Planning Error: java.lang.RuntimeException: Error compiling query: org.voltdb.VoltTypeException: tryToMakeCompatible: The provided value: (2015-11-111 10:10:10) of type: java.lang.String is not a match or is out of range for the target parameter type: org.voltdb.types.TimestampType (Stack trace has been written to the log.)
-Unexpected Ad Hoc Planning Error: java.lang.RuntimeException: Error compiling query: org.voltdb.VoltTypeException: tryToMakeCompatible: The provided value: (2015-11-11 10:10:10.) of type: java.lang.String is not a match or is out of range for the target parameter type: org.voltdb.types.TimestampType (Stack trace has been written to the log.)
+SQL error while compiling query: tryToMakeCompatible: The provided value: (20150-11-11 10:10:10) of type: java.lang.String is not a match or is out of range for the target parameter type: org.voltdb.types.TimestampType (Stack trace has been written to the log.)
```
Comment (yzhang1991) [inline tests/sqlcmd/baselines/simple/parsedates.errbaseline]: tryToMakeCompatible is also not needed. You can remove it from the code. Look at `ParameterConverter.java`

File: src/frontend/org/voltdb/compiler/PlannerTool.java (hunk 1 above)
Comment (mreading) [reply src/frontend/org/voltdb/compiler/PlannerTool.java]: While running the TestFixedSQLSuite tests, I encountered this error without this addition:

    [junit] Testcase: testFixedTickets-localCluster-1-1-HSQLDB took 1.167 sec
    [junit] 	FAILED
    [junit] Expected the statement "SELECT NUM + 11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111.1111E1 FROM R1" to throw an exception matching the pattern "java.lang.NumberFormatException", but instead it threw an exception containing "SQL error while compiling query: null (Stack trace has been written to the log.)".
    [junit] junit.framework.AssertionFailedError: Expected the statement "SELECT NUM + 11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111.1111E1 FROM R1" to throw an exception matching the pattern "java.lang.NumberFormatException", but instead it threw an exception containing "SQL error while compiling query: null (Stack trace has been written to the log.)".
    [junit] 	at org.voltdb.regressionsuites.RegressionSuite.verifyProcFails(RegressionSuite.java:1213)
    [junit] 	at org.voltdb.regressionsuites.RegressionSuite.verifyStmtFails(RegressionSuite.java:1188)
    [junit] 	at org.voltdb.regressionsuites.TestFixedSQLSuite.subTestENG7480(TestFixedSQLSuite.java:2683)
    [junit] 	at org.voltdb.regressionsuites.TestFixedSQLSuite.testFixedTickets(TestFixedSQLSuite.java:899)

Without printing the java exception error, the overall error messages makes no sense - see the "...query: null" - so I chose to keep the original format for NumberFormatException cases

Comment (mreading) [reply src/frontend/org/voltdb/compiler/PlannerTool.java]: (you have to scroll far to the right)

File: tests/sqlcmd/baselines/simple/parsedates.errbaseline (hunk 2 above)
Comment (mreading) [reply tests/sqlcmd/baselines/simple/parsedates.errbaseline]: Should I remove all VoltTypeException instances of tryToMakeCompatible in that file?

File: src/frontend/org/voltdb/compiler/PlannerTool.java [hunk 3]
```diff
@@ -158,10 +158,10 @@ public synchronized CompiledPlan planSqlCore(String sql, StatementPartitioning p
                 logException(e, "Error compiling query");
                 loggedMsg = " (Stack trace has been written to the log.)";
             }
-            if (e instanceof NumberFormatException) {
-                throw new RuntimeException("Error compiling query: " + e.toString() + loggedMsg, e);
+            if (e.getMessage() != null) {
+                throw new RuntimeException("SQL error while compiling query: " + e.getMessage() + loggedMsg, e);
             }
-            throw new RuntimeException("SQL error while compiling query: " + e.getMessage() + loggedMsg, e);
+            throw new RuntimeException("Error compiling query: " + e.toString() + loggedMsg, e);
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/compiler/PlannerTool.java]: Why is this message different? Should they be both "SQL error while..." or just both "Error compiling query"?

Comment (mreading) [reply src/frontend/org/voltdb/compiler/PlannerTool.java]: The ticket suggested to say "SQL error while compiling query", but the original version was "Error compiling query". I only added the change to the code I added, but I can make it consistent across. Which one do you prefer?

**Pull request merged**